app/javascript/components/mapping-to-domains/UploadVocabulary.jsx

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect, useState } from "react";
+import createVocabulary from "../../services/createVocabulary";
+import FileInfo from "../mapping/FileInfo";
+
+const UploadVocabulary = (props) => {
+  /**
+   * The uploaded file whixh will contain the specification of
+   * a vocabulary in a concept scheme json-ld format
+   */
+  const [file, setFile] = useState(null);
+  const [fileContent, setFileContent] = useState("");
+  const [name, setName] = useState("");
+
+  /**
+   * Update the files in the redux store (main application state) when
+   * the input changes (after the user selects file/s)
+   */
+  const handleFileChange = (event) => {
+    setFile(event.target.files[0]);
+  };
+
+  /**
+   * After the sate of the file is changed, read it if there's any valid
+   * file
+   */
+  useEffect(() => {
+    if (file != null) {
+      readFileContent();
+    }
+  }, [file]);
+
+  /**
+   * File content to be displayed after
+   * file upload is complete
+   *
+   * @returns {React.Fragment}
+   */
+  const fileData = () => {
+    return file != null ? (
+      <FileInfo selectedFile={file} key={Date.now() + file.lastModified} />
+    ) : (
+      ""
+    );
+  };
+
+  /**
+   * Transform the file content to be able to send it to the backend
+   */
+  const readFileContent = () => {
+    const reader = new FileReader();
+
+    /**
+     * When reading the file content
+     */
+    reader.onload = () => {
+      /// Get the content of the file
+      let content = reader.result;
+      setFileContent(content);
+    };
+
+    /**
+     * Update callback for errors
+     */
+    reader.onerror = function (e) {
+      setErrors("File could not be read! Code: " + e.target.error.code);
+    };
+
+    reader.readAsText(file);
+  };
+
+  /**
+   * Send the file with the vocabulary to the backend
+   */
+  const handleCreateVocabulary = (event) => {
```

Comment:
This handler doesn't appear to account for possible errors. Also, let's validate the file content before attempting to create a vocabulary in order to save a trip to the backend.

Reply:
Done. See [screenshot](https://user-images.githubusercontent.com/6996951/94058356-45327a80-fdb7-11ea-9047-cbb427cf1cd6.gif)